### PR TITLE
[x/programs] Improve simulator logs

### PR DESCRIPTION
**Pull request by dboehm-avalabs**
Simulator logs currently escape new lines, which makes reading them difficult.  Fixes for the output in rust tests as well as the simulator.  Our simulator based tests already print any returned errors, so making no stdout printing the default.

### CODE REVIEW

File: x/programs/cmd/simulator/src/lib.rs [hunk 1]
```diff
@@ -186,7 +186,15 @@ pub struct PlanResponse {
...
-    pub error: Option<String>,
+    pub error: Option<PlanError>,
+}
+
+#[derive(Serialize, Deserialize)]
+pub struct PlanError(String);
+impl core::fmt::Debug for PlanError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}", self.0)
+    }
```
Comment (richardpringle) [inline x/programs/cmd/simulator/src/lib.rs:197]: Do you need this? 

It doesn't appear to be doing anything

Comment (dboehm-avalabs) [reply x/programs/cmd/simulator/src/lib.rs:197]: yeah, without it the default behavior is to escape new lines.

`---- tests::mint_and_transfer stdout ----
thread 'tests::mint_and_transfer' panicked at x/programs/rust/examples/token/src/lib.rs:448:9:
error: Some("program execution failed: \n, err: error while executing at wasm backtrace:\n    0: 0x32619 - <unknown>!std::thread::local::LocalKey<T>::try_with::h217d06e356708221\n    1: 0x3253a - <unknown>!std::thread::local::LocalKey<T>::with::h505f088a73c802e5\n    2: 0x32cde - <unknown>!std::thread::local::LocalKey<core::cell::RefCell<T>>::with_borrow_mut::h42f856ca1bb87954\n    3: 0x318e9 - <unknown>!alloc\nnote: using the `WASMTIME_BACKTRACE_DETAILS=1` environment variable may show more debugging information\n\nCaused by:\n    wasm trap: all fuel consumed by WebAssembly")
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace` vs

`---- tests::mint_and_transfer stdout ----
thread 'tests::mint_and_transfer' panicked at x/programs/rust/examples/token/src/lib.rs:448:9:
error: Some(program execution failed:
, err: error while executing at wasm backtrace:
    0: 0x32619 - <unknown>!std::thread::local::LocalKey<T>::try_with::h217d06e356708221
    1: 0x3253a - <unknown>!std::thread::local::LocalKey<T>::with::h505f088a73c802e5
    2: 0x32cde - <unknown>!std::thread::local::LocalKey<core::cell::RefCell<T>>::with_borrow_mut::h42f856ca1bb87954
    3: 0x318e9 - <unknown>!alloc
note: using the `WASMTIME_BACKTRACE_DETAILS=1` environment variable may show more debugging information

Caused by:
    wasm trap: all fuel consumed by WebAssembly)
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace`

Comment (dboehm-avalabs) [reply x/programs/cmd/simulator/src/lib.rs:197]: Screenshot 2024-05-29 at 12 28 20 PM
Screenshot 2024-05-29 at 12 29 03 PM

**Review by iFrostizz — COMMENTED** [review]
I have a suggestion that would slightly improve the error message (that is already looking really good) so that it's now:

```sh
thread 'tests::increment' panicked at x/programs/rust/examples/counter/src/lib.rs:155:9:
error: Some(program execution failed: BBBBBBBBBB, err: this is a sample error)
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

with a backtrace:

```sh
---- tests::increment stdout ----
thread 'tests::increment' panicked at x/programs/rust/examples/counter/src/lib.rs:156:9:
error: Some(program execution failed, err: error while executing at wasm backtrace:
    0: 0x342a2 - counter.wasm!__rust_start_panic
    1: 0x34245 - counter.wasm!rust_panic
    2: 0x34237 - counter.wasm!std::panicking::rust_panic_with_hook::h32c80a64fe4de396
    3: 0x3392c - counter.wasm!std::panicking::begin_panic_handler::{{closure}}::hd496964d114e98b9
    4: 0x3388f - counter.wasm!std::sys_common::backtrace::__rust_end_short_backtrace::h0d4686a7fe3981a4
    5: 0x33ec6 - counter.wasm!rust_begin_unwind
    6: 0x34a9d - counter.wasm!core::panicking::panic_fmt::hc7427f902a13f1a9
    7: 0x35e5a - counter.wasm!core::panicking::panic_display::h57441543e669293c
    8: 0x35f57 - counter.wasm!core::panicking::panic_explicit::h2a7b86e0d5853ff3
    9: 0x65c9 - counter.wasm!counter::inc::panic_cold_explicit::hc58cd8ef96c1d50c
   10: 0x659b - counter.wasm!counter::inc::h4fb50fd00d7b3d16
   11: 0x67bc - counter.wasm!inc
note: using the `WASMTIME_BACKTRACE_DETAILS=1` environment variable may show more debugging information

Caused by:
    wasm trap: wasm `unreachable` instruction executed)
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

(Github does not support writing code changes on unchanged code)

```go
	// execute the action
	resp, err := programExecuteAction.Execute(ctx, nil, db, 0, codec.EmptyAddress, programTxID)
	if err != nil {
		response := multilineOutput(resp)
		if len(response) > 0 {
			err = fmt.Errorf("program execution failed: %s, err: %w", response, err)
		} else {
			err = fmt.Errorf("program execution failed, err: %w", err)
		}
		return ids.Empty, nil, 0, err
	}
	
...

func multilineOutput(resp [][]byte) (response string) {
	for _, res := range resp {
		response += string(res)
	}
	return response
}
```

File: x/programs/cmd/simulator/src/lib.rs (hunk 1 above)
Comment (richardpringle) [reply x/programs/cmd/simulator/src/lib.rs:197]: Oh... I see what's happening, it's `Display` formatting instead of debug formatting

**Review by richardpringle — APPROVED** [review]
I don't like doing `Debug` as `Display`, but this is better than it was. I can look into fixing it further

**Pull request merged**